models/yang/sonic/sonic-neigh.yang

```diff
@@ -0,0 +1,53 @@
+module sonic-neigh {
+    namespace "http://github.com/Azure/sonic-neigh";
+    prefix sneigh;
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC yang for NEIGH_TABLE";
+
+    revision 2019-10-10 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-neigh {
+
+        container NEIGH_TABLE {
+
+            list NEIGH_TABLE_LIST {
+
+                key "ifname ip";
+
+                leaf ifname {
+                    type string;
```

Comment:
This should be leafref to INTERFACE table ?

Reply:
The INTERFACE_LIST in sonic-interface.yang has a '**portname**' field, which is a leaf reference to '**ifname**' defined in sonic-port.yang.  However, the **ifname** has a pattern that allows Ethernet only, where in our case the interface can be Ethernet/Vlan/PortChannel/Management. 

There can be different ways to resolve this issue: 
either extend sonic-port.yang and allow different interface types or I can add a pattern for different interface types in this sonic yang or we can leave as it is.  As it is a **Get** operation, where data is expected to come in right form from the DB, I am more inclined towards the third approach.

@joyas-joseph  and @kwangsuk  please suggest

Reply:
@dutta-partha There is no single reference point for all different interfaces types in this YANG as Obaid mentioned. I think that the string type would be ok since it is config false anyway.

Reply:
Probably, this needs union here.